soc/nxp/imx/imx9/imx95/m7/CMakeLists.txt

```diff
@@ -5,4 +5,4 @@ zephyr_include_directories(.)
 zephyr_library()
 zephyr_library_sources(soc.c)
 
-set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/arm/cortex_m/scripts/linker.ld CACHE INTERNAL "")
+set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
```

Comment:
@AndreHeinemans-NXP you should split this commit in more patches.
Keep the i.MX95 M7 support for `openamp_rsc_table` sample separate that the changes to the sample.

For example, here's how I'm doing it: https://github.com/zephyrproject-rtos/zephyr/pull/79220/commits
I'm not saying is the best approach, but I believe is easier to review.